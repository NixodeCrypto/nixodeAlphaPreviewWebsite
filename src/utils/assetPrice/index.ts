import assetPrice from './assetPrice';

export default assetPrice;
