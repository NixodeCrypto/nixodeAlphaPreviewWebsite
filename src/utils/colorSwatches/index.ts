import colorSwatches from './colorSwatches';

export * from './colorSwatches';
export default colorSwatches;
