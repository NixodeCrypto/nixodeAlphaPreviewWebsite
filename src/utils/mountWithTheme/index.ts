import mountWithTheme from './mountWithTheme';

export default mountWithTheme;
