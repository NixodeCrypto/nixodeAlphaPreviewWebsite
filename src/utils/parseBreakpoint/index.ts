import parseBreakpoint from './parseBreakpoint';

export default parseBreakpoint;
