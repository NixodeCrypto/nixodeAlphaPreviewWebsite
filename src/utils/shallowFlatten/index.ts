import shallowFlatten from './shallowFlatten';

export default shallowFlatten;
