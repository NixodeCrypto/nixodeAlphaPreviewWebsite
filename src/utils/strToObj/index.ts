import strToObj from './strToObj';

export default strToObj;
