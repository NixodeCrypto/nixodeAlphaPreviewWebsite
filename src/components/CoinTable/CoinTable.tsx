/* @jsxImportSource @emotion/react */
import { useState, useEffect } from 'react';
import styled from '@emotion/styled';
import SVG from 'react-inlinesvg';
import { css } from '@emotion/react';
import Image from '@/components/Image';
import Flex from '@/components/Flex';
import Body from '@/components/Body';
import Box from '@/components/Box';
import { token, mq, assetPrice } from '@/utils';
import { space, border, layout, typography } from '@/system';

export interface IProps {
  tickerData: Record<string, any>;
  extended?: boolean;
  timeResolution?: string;
}

const Table = styled.table`
  position: relative;
  width: 100%;
  border-collapse: collapse;
  ${mq('md')} {
    border-style: hidden;
    border-radius: ${token.radii('sm')};
    box-shadow: 0 0 0 2px ${token.colors('grey.100')};
    border-color: ${token.colors('grey.100')};
  }
  ${mq('xss')} {
    display: block;
    overflow-x: scroll;
  }
  ${mq('xs')} {
    display: table;
    overflow-x: auto;
  }
`;

const Tr = styled.tr`
  transition: ${token.transition('standard')};
  ${(props) =>
    !props.header &&
    css`
      &:hover {
        background-color: ${token.colors('grey.50')};
        transition: ${token.transition('standard')};
      }
    `}
  ${space};
  ${border};
`;

const Thead = styled.thead`
  ${layout};
`;

const Td = styled.td`
  ${space};
  ${layout};
  vertical-align: middle;
  font-family: ${token.fonts('text')};
`;

const Th = styled.th`
  font-family: ${token.fonts('title')};
  font-size: ${token.fontSizes('captionLg')};
  font-weight: ${token.fontWeights('regular')};
  padding-top: ${token.space('sm')};
  padding-bottom: ${token.space('sm')};
  color: ${token.colors('grey.900')};
  ${typography};
  ${space};
  ${layout};
`;

const percentFormatter = (percentage: number) => {
  if (percentage === 0) {
    return `${percentage.toLocaleString(undefined, {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    })}%`;
  }
  return percentage.toString().substr(0, 1) === '-'
    ? `${percentage.toLocaleString(undefined, {
        minimumFractionDigits: 2,
        maximumFractionDigits: 2,
      })}%`
    : `+${percentage.toLocaleString(undefined, {
        minimumFractionDigits: 2,
        maximumFractionDigits: 2,
      })}%`;
};

const CoinTable = (props: IProps) => {
  const { tickerData, extended, timeResolution = '24h' } = props;

  return (
    <Table>
      <Thead
        display={{ xss: 'none', sm: 'table-header-group' }}
        data-testid="TableHeader"
      >
        <Tr borderTop="sm" borderColor="grey.100" header>
          <Th textAlign="left" pl="sm">
            Name
          </Th>
          <Th textAlign="right" pr="md">
            Price
          </Th>
          <Th textAlign="right" pr="md">
            Change
          </Th>
          <Th textAlign="center" px="sm">
            Chart (7d)
          </Th>
          {extended && (
            <>
              <Th
                textAlign="right"
                pr="md"
                display={{ xss: 'none', md: 'table-cell' }}
              >
                Volume (24h)
              </Th>
              <Th
                textAlign="right"
                pr="md"
                display={{ xss: 'none', md: 'table-cell' }}
              >
                Market cap
              </Th>
              <Th
                textAlign="right"
                pr="md"
                display={{ xss: 'none', md: 'table-cell' }}
              >
                Supply
              </Th>
            </>
          )}
        </Tr>
      </Thead>
      <tbody>
        {tickerData &&
          tickerData.map((i: any, idx: number) => (
            <Tr
              key={i.id}
              borderTop={!extended ? 'sm' : idx === 0 && 'sm'}
              borderColor="grey.100"
            >
              <Td px="sm" py="sm" data-testid="generalData">
                <Flex alignItems="center">
                  <Image src={i.img} alt={i.name} width="sm" mr="xs" />
                  <Flex
                    flexDirection={{ xss: 'column', sm: 'row' }}
                    alignItems={{ md: 'center' }}
                  >
                    <Body mr={{ sm: 'sm' }}>{i.name}</Body>
                    <Body color="grey.800">{i.symbol}</Body>
                  </Flex>
                </Flex>
              </Td>
              <Td px="md" py="sm" width="md">
                <Flex
                  alignItems="flex-end"
                  justifyContent="flex-end"
                  flexDirection="column"
                >
                  <Body textAlign="right">
                    US${assetPrice(i.quotes.USD.price)}
                  </Body>
                  <Body
                    display={{ sm: 'none' }}
                    color={
                      i.quotes.USD[`percent_change_${timeResolution}`] < 0
                        ? 'red.800'
                        : 'green.800'
                    }
                  >
                    {percentFormatter(
                      i.quotes.USD[`percent_change_${timeResolution}`],
                    )}
                  </Body>
                </Flex>
              </Td>
              <Td
                display={{ xss: 'none', sm: 'table-cell' }}
                px="md"
                py="sm"
                width="md"
              >
                <Flex alignItems="flex-end" flexDirection="column">
                  <Body
                    color={
                      i.quotes.USD[`percent_change_${timeResolution}`] < 0
                        ? 'red.800'
                        : 'green.800'
                    }
                  >
                    {percentFormatter(
                      i.quotes.USD[`percent_change_${timeResolution}`],
                    )}
                  </Body>
                </Flex>
              </Td>
              <Td
                display={{ xss: 'none', sm: 'table-cell' }}
                px="md"
                py="sm"
                width="xl"
                minWidth="xl"
              >
                <Box
                  css={css`
                    & * {
                      fill: none;
                      stroke-width: 0.4rem;
                    }
                    & polyline {
                      stroke: ${token.colors('grey.400')};
                    }
                  `}
                >
                  <SVG
                    src={`${process.env.NEXT_PUBLIC_GRAPH_API}/currency/chart/${i.id}/7d/chart.svg`}
                    width="100%"
                    height="24px"
                  />
                </Box>
              </Td>
              {extended && (
                <>
                  <Td
                    px="md"
                    width="md"
                    py="sm"
                    display={{ xss: 'none', md: 'table-cell' }}
                  >
                    <Flex justifyContent="flex-end" suppressHydrationWarning>
                      US$
                      {Intl.NumberFormat('en-US', {
                        notation: 'compact',
                        maximumFractionDigits: 1,
                      }).format(i.quotes.USD.volume_24h)}
                    </Flex>
                  </Td>
                  <Td
                    px="md"
                    width="md"
                    py="sm"
                    display={{ xss: 'none', md: 'table-cell' }}
                  >
                    <Flex justifyContent="flex-end" suppressHydrationWarning>
                      US$
                      {Intl.NumberFormat('en-US', {
                        notation: 'compact',
                        maximumFractionDigits: 1,
                      }).format(i.quotes.USD.market_cap)}
                    </Flex>
                  </Td>
                  <Td
                    px="md"
                    width="md"
                    py="sm"
                    display={{ xss: 'none', md: 'table-cell' }}
                  >
                    <Flex justifyContent="flex-end" suppressHydrationWarning>
                      {Intl.NumberFormat('en-US', {
                        notation: 'compact',
                        maximumFractionDigits: 1,
                      }).format(i.circulating_supply)}
                    </Flex>
                  </Td>
                </>
              )}
            </Tr>
          ))}
      </tbody>
    </Table>
  );
};

CoinTable.displayName = 'CoinTable';
export default CoinTable;
