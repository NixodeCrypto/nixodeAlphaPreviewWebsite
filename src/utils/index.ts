import variant from '@/utils/variant';
import colorSwatches from '@/utils/colorSwatches';
import token from '@/utils/token';
import mq from '@/utils/mq';
import parseBreakpoint from '@/utils/parseBreakpoint';
import mountWithTheme from '@/utils/mountWithTheme';
import strToObj from '@/utils/strToObj';

export {
  token,
  variant,
  colorSwatches,
  mq,
  parseBreakpoint,
  mountWithTheme,
  strToObj,
};
