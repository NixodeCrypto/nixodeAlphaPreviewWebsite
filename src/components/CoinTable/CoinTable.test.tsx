import { ReactWrapper } from 'enzyme';
import CoinTable from '.';
import { mountWithTheme } from '@/utils';

let sampleData: Record<string, any>;
let wrapper: ReactWrapper;

beforeAll(() => {
  sampleData = [
    {
      _id: '61aed321c05b119295c88fe9',
      id: 'btc-bitcoin',
      name: 'Bitcoin',
      img: 'https://static.coinpaprika.com/coin/btc-bitcoin/logo.png',
      symbol: 'BTC',
      rank: 1,
      circulating_supply: 18912744,
      total_supply: 18912744,
      max_supply: 21000000,
      beta_value: 0.937577,
      first_data_at: '2010-07-17T00:00:00Z',
      last_updated: '2021-12-27T22:57:08Z',
      quotes: {
        USD: {
          price: 51036.656169985,
          volume_24h: 27135932523.212,
          volume_24h_change_24h: 12.08,
          market_cap: 9.65243212758e11,
          market_cap_change_24h: 0.23,
          percent_change_15m: 0.01,
          percent_change_30m: -0.34,
          percent_change_1h: -0.11,
          percent_change_6h: -0.87,
          percent_change_12h: 0.35,
          percent_change_24h: 0.23,
          percent_change_7d: 8.84,
          percent_change_30d: -6.63,
          percent_change_1y: 92.41,
          ath_price: 68692.137036932,
          ath_date: '2021-11-10T16:51:15Z',
          percent_from_price_ath: -25.7,
        },
      },
    },
  ];
  wrapper = mountWithTheme(<CoinTable tickerData={sampleData} />);
});

describe('components/CoinTable', () => {
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe('Table Head', () => {
    it('Name Th Tag Found', () => {
      expect(
        wrapper
          .find({ 'data-testid': 'TableHeader' })
          .at(0)
          .text()
          .split(/(?=[A-Z])/),
      ).toContain('Name');
    });
    it('Price Th Tag Found', () => {
      expect(
        wrapper
          .find({ 'data-testid': 'TableHeader' })
          .at(0)
          .text()
          .split(/(?=[A-Z])/),
      ).toContain('Price');
    });
    it('Change Th Tag Found', () => {
      expect(
        wrapper
          .find({ 'data-testid': 'TableHeader' })
          .at(0)
          .text()
          .split(/(?=[A-Z])/),
      ).toContain('Change');
    });
    it('Chart (7d) Th Tag Found', () => {
      expect(
        wrapper
          .find({ 'data-testid': 'TableHeader' })
          .at(0)
          .text()
          .split(/(?=[A-Z])/),
      ).toContain('Chart (7d)');
    });
  });
  describe('Table Body', () => {
    describe('General Data', () => {
      it('Icon src test', () => {
        expect(
          wrapper
            .find({ 'data-testid': 'generalData' })
            .at(0)
            .find('Image')
            .prop('src'),
        ).toEqual(sampleData[0].img);
      });
      it('Icon alt test', () => {
        expect(
          wrapper
            .find({ 'data-testid': 'generalData' })
            .at(0)
            .find('Image')
            .prop('alt'),
        ).toEqual(sampleData[0].name);
      });
      it('Name rendering test', () => {
        expect(
          wrapper
            .find({ 'data-testid': 'generalData' })
            .at(0)
            .text()
            .includes('Bitcoin'),
        ).toBeTruthy();
      });
      it('Symbol rendering test', () => {
        expect(
          wrapper
            .find({ 'data-testid': 'generalData' })
            .at(0)
            .text()
            .includes('BTC'),
        ).toBeTruthy();
      });
    });
    describe('Market Data', () => {
      it('Price rendering test', () => {
        expect(wrapper.text().includes('$51,036.66')).toBeTruthy();
      });
      it('24hr change rendering test', () => {
        expect(wrapper.text().includes('+0.23%')).toBeTruthy();
      });
    });
  });
});
