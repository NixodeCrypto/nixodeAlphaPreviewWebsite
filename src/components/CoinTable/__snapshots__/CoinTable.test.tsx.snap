// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CoinTable matches snapshot 1`] = `ReactWrapper {}`;
